extensions/ql-vscode/src/extension.ts

```diff
@@ -774,10 +775,15 @@ async function activateWithInstalledDistribution(
   void dbm.loadPersistedState();
 
   ctx.subscriptions.push(dbm);
+
+  void extLogger.log("Initializing language context.");
+  const languageContext = new LanguageContextStore(app);
```

Comment:
Are these changes meant to be behind a feature flag or did I imagine that?

Reply:
I thought they were, but I cannot see one before this PR either?

Reply:
The "Languages" dropdown is feature-flagged in the UI here: https://github.com/github/vscode-codeql/blob/6aab4b4090600ce4d6ebd9183feb2cd215b809ac/extensions/ql-vscode/package.json#L1080

So it might make sense to feature-flag this code too 🎏

Reply:
:+1: I have had the canary flag as a "standard" flag, so I didnt even think about it :doh:

How do we normally flag this? If we just omit the creation of the languageContext then we'll have a lot of places where it is suddenly undefined, which seems more risky than having it in the first place?

Reply:
For the queries panel, we check for the flag before initializing: https://github.com/github/vscode-codeql/blob/2e45dcd7b9a0aa01deeb1605e18d8eaa6515c14e/extensions/ql-vscode/src/queries-panel/queries-module.ts#L29-L33 

But in this case, I think it's fairly harmless to create it without feature-flagging 🤔 I don't really know what's "best" or "safest" 😅

Reply:
Ideally no code that does language filtering stuff should be executed if the flag is off. But I think it's okay to be lenient - it's all about us having enough confidence when releasing rather than sticking to specific rules.

Reply:
I dont see a good way of feature-flagging this. But happy to discuss more sync.